docs/_chapters/230-manifest-annotations.md

```diff
@@ -1,136 +1,286 @@
+### @Requirement & @Capability Annotations
 
-Meet the `@RequireCapability` and `@ProvideCapability` headers. 
+Though Java class files contain enough information to find code dependencies, there are many dependencies that are indirect. OSGi _extenders_ for instance are often a requirement to make a bundle function correctly but often client bundles have no code dependency on the extender. For example, Declarative Services (DS) went out of its way to allow components to be Plain Old Java Objects (POJO). The result is that resolving a closure of bundles starting from a DS client bundle would not drag in the Service Component Runtime (SCR), resulting in a satisfied but rather idle closure.
 
-These annotations can be applied to a type or any other annotation. If applied to a type then the annotation's requirement or capability will be added to the manifest. If applied to an annotation then this annotation is a customized annotation. Nothing happens until the annotated annotation is used. When this annotation is applied somewhere, bnd will automatically add the requirement or capability to the manifest.
+The solution was to describe the requirement for the runtime SCR dependency using [Requirements and Capabilities](https://osgi.org/specification/osgi.core/7.0.0/framework.module.html#framework.module.dependencies). But again, writing these complex clauses in the manifest by hand is both error prone and painful.
 
-### RequireCapability
+The [`@Requirement`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Requirement) and [`@Capability`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Capability) annotations were designed to address this issue. These annotations can be used to create [*custom bundle annotations*](#custom-bundle-annotations), described later on. Let's discuss the DS example.
 
-The `@RequireCapability` annotation creates a clause in the Require-Capability manifest header. The annotation has the following fields (for more details consult the Javadoc):
+Recent DS specifications require implementations to provide the following capability:
 
-* `value` – (String) Anything in the value field is appended after the calculated header.
-* `ns` – (String) The namespace of the requirement
-* `effective` – (String default `resolve`) The effective time
-* `filter` – The filter directive. There is a handy macro [${frange;<version>][1] that turns a version into a filter expression for a filter range.  
-* `resolution` – (`mandatory` | `optional`) The resolution of the requirement.
+```
+Provide-Capability: osgi.extender;
+    osgi.extender="osgi.component";
+    version:Version="x";
```

Comment:
`x` is not a real version number and the example should be use a proper version number.